src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1703,13 +1703,45 @@ def aks_agentpool_add(cmd, client, resource_group_name, cluster_name, nodepool_n
                       min_count=None,
                       max_count=None,
                       enable_cluster_autoscaler=False,
+                      initial_node_taints=None,
+                      priority="Regular",
+                      eviction_policy="Delete",
                       no_wait=False):
     instances = client.list(resource_group_name, cluster_name)
     for agentpool_profile in instances:
         if agentpool_profile.name == nodepool_name:
             raise CLIError("Node pool {} already exists, please try a different name, "
                            "use 'aks nodepool list' to get current list of node pool".format(nodepool_name))
 
+    taints_array = []
+
+    if initial_node_taints is not None:
+        for taint in initial_node_taints.split(','):
+            try:
+                taint = taint.strip()
+                taints_array.append(taint)
+            except ValueError:
+                raise CLIError('Taint is not well-formed.')
```

Comment:
Suggest updating error to "Taint does not match allowed values. Expect value such as "special=true:NoSchedule"." or something other than not well-formed.